File: 25/12/3074.cpp
#include <algorithm>
#include <numeric>
#include <vector>

class Solution {
public:
    int minimumBoxes(std::vector<int>& apple, std::vector<int>& capacity) {
        int sum = std::accumulate(apple.begin(),apple.end(),0);
        std::sort(capacity.begin(),capacity.end(),[](int a,int b){return a>b;});
        int ans = 0;
        for(auto c : capacity){
            sum -= c;
            ans++;
            if (sum <= 0) {
                break;
            }
        }
        return ans;;
    }
};

File: 25/12/go/3074.go
package A

import "slices"

func minimumBoxes(apple []int, capacity []int) int {
	sum := 0
	for _, i := range apple {
		sum += i
	}
	slices.SortFunc(capacity, func(a, b int) int { return b - a })
	ans := 0
	for _, i := range capacity {
		sum -= i
		ans++
		if sum <= 0 {
			break
		}
	}
	return ans
}
